Easydict/Feature/Service/Gemini/GeminiTranslateType.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  GeminiTranslateType.swift
+//  Easydict
+//
+//  Created by Jerry on 2024-01-02.
+//  Copyright © 2024 izual. All rights reserved.
+//
+
+import Foundation
+
+struct GeminiTranslateType: RawRepresentable {
+    var rawValue: String
+
+    static let unsupported = GeminiTranslateType(rawValue: "unsupported")
+
+    // https://ai.google.dev/available_regions
+    static let supportedTypes: [Language: [Language]] = [
+        .arabic: [.arabic, .bengali, .bulgarian, .simplifiedChinese, .traditionalChinese, .croatian, .czech, .danish, .dutch, .english, .estonian, .finnish, .french, .german, .greek, .hebrew, .hindi, .hungarian, .indonesian, .italian, .japanese, .korean, .latvian, .lithuanian, .norwegian, .polish, .portuguese, .romanian, .russian, .serbian, .slovak, .slovenian, .spanish, .swedish, .thai, .turkish, .ukrainian, .vietnamese],
```

Comment:
Do we have a better idea here? The giant array literal may crash or slow down the compiler due to a known Swift compiler bug.

Reply:
The good thing about Gemini is that all the languages are interoperable, so I'm thinking if it is possible to achieve the following:

Instead of
```
.chinese: [.chinese, .english, .french]
.english: [.chinese, .english, .french]
.french: [.chinese, .english, .french]
```
Is this possible:
```
.chinese, .english, .french: [.chinese, .english, .french]
```

Reply:
我没有看懂这里的代码是在做什么，按理说 Gemini 应该是支持所有语言直接相互理解转换的，这样我们根本不用处理。